pkg/registry/populator.go

```diff
@@ -20,22 +20,22 @@ type Dependencies struct {
 
 // DirectoryPopulator loads an unpacked operator bundle from a directory into the database.
 type DirectoryPopulator struct {
-	loader          Load
-	graphLoader     GraphLoader
-	querier         Query
-	imageDirMap     map[image.Reference]string
-	overwriteDirMap map[string]map[image.Reference]string
-	overwrite       bool
+	loader            Load
+	graphLoader       GraphLoader
+	querier           Query
+	imageDirMap       map[image.Reference]string
+	overwrittenImages map[string][]string
+	overwrite         bool
 }
 
-func NewDirectoryPopulator(loader Load, graphLoader GraphLoader, querier Query, imageDirMap map[image.Reference]string, overwriteDirMap map[string]map[image.Reference]string, overwrite bool) *DirectoryPopulator {
+func NewDirectoryPopulator(loader Load, graphLoader GraphLoader, querier Query, imageDirMap map[image.Reference]string, overwrittenImages map[string][]string, overwrite bool) *DirectoryPopulator {
 	return &DirectoryPopulator{
-		loader:          loader,
-		graphLoader:     graphLoader,
-		querier:         querier,
-		imageDirMap:     imageDirMap,
-		overwriteDirMap: overwriteDirMap,
-		overwrite:       overwrite,
+		loader:            loader,
+		graphLoader:       graphLoader,
+		querier:           querier,
+		imageDirMap:       imageDirMap,
+		overwrittenImages: overwrittenImages,
```

Comment:
nit: do we need a separate `overwrite` variable anymore? Isn't `len(overwrittenImages) == 0` sufficient?